### ISSUE TITLE
Error for byte literal

### ISSUE THREAD

**Issue by andy-hanson** [intellij-rust/intellij-rust#1739]
```rust
fn main() {
	let _: u8 = b'a';
}
```

This code isn't an error when using the command line compiler, but does cause an error inside the editor:
```
Mismatched types E0308
expected u8, found char
```

There is no error for byte string literals: `let _: &[u8] = b"abc";` works.
